lib/admin.rb

```diff
@@ -0,0 +1,116 @@
+require_relative 'reservation'
+require 'date'
+require 'pry'
+
+module Hotel
+  class Admin
+    attr_reader :rooms, :reservations, :blocks
+
+    def initialize
+      @reservations = []
+      @rooms = room_list
+      @blocks = []
+    end
+
+    def room_list
+      rooms = []
+      (1..20).each { |i| rooms << i }
+      return rooms
+    end
+
+    def book_room(reservation)
+      if reservation.class != Reservation
+        raise ArgumentError.new("Can only use reservation instance to book a room #{reservation}")
+      end
+
+      free_rooms_for_dates(reservation.dates).pop
+    end
+
+    def add_reservation(reservation)
+      if reservation.class != Reservation
+        raise ArgumentError.new("Not a valid reservation instance #{reservation}")
+      elsif free_rooms_for_dates(reservation.dates).length == 0
+        raise ArgumentError.new("No available rooms for these dates #{reservation.dates}")
+      end
+
+      @reservations << reservation
+      book_room(reservation)
+    end
+
+    def create_block_of_rooms(reservation, num_rooms)
```

Comment:
What about:
1.  Creating a block with a discounted room rate.
1.  Is a block a `Reservation`?  
1.  Is a room really reserved, if it's in the block, or is it just being held for later?